app/controllers/api/reports_controller.rb

```diff
@@ -14,7 +14,7 @@ def favorite
     end
 
     def unfavorite
-      current_user.favorite_reports.where(id: params[:id].to_i).destroy_all
+      current_user.favorites.where(entity: params[:id].to_i, entity_type: 'GrdaWarehouse::WarehouseReports::ReportDefinition').destroy_all
```

Comment:
Looks good to me. Good find. I think this was my first feature I did at GR when I didn't know rails at all 😬 👍